myframe_events.cpp

```diff
@@ -1013,26 +1016,60 @@ void MyFrame::GuideButtonClick(bool interactive, const wxString& context)
                 if (pSecondaryMount)
                     pSecondaryMount->ClearCalibration();
             }
+            else
+                return;
         }
 
-        if (interactive && pPointingSource && pPointingSource->IsConnected() && pPointingSource->CanReportPosition())
+        if (interactive && pPointingSource && pPointingSource->IsConnected() &&
+            pPointingSource->CanReportPosition())
         {
-            bool proceed = true;
-            bool error = pPointingSource->PreparePositionInteractive();
 
-            if (!error && fabs(pPointingSource->GetDeclination()) > Scope::DEC_COMP_LIMIT && !TheScope()->IsCalibrated() )
+            if (pPointingSource->PreparePositionInteractive())
+                return;
+
+            if (!TheScope()->IsCalibrated())
             {
-                proceed = ConfirmDialog::Confirm(
-                    _("Calibration this far from the celestial equator will be error-prone.  For best results, calibrate at a declination of -20 to +20."),
-                    "/highdec_calibration_ok", _("Confirm Calibration at Large Declination")
-                    );
+                double ra;
+                double dec;
+                double lst;
+                if (!pPointingSource->GetCoordinates(&ra, &dec, &lst))
+                {
+                    bool calHere = true;
+                    double ha = norm(lst - ra, -12.0, 12.0);
+                    if ((fabs(dec) > 20 && dec < degrees(Scope::DEC_COMP_LIMIT)) || fabs(ha) > 3)
+                        calHere = ConfirmDialog::Confirm(
+                        _("Scope isn't pointing in recommended sky area - run the Calibration Assistant to improve results"),
+                        "/v2_highdec_calibration_ok", _("Calibrate here"), _("Calibration Assistant...")
+                        );
+                    else if (dec > 60)
+                        calHere = ConfirmDialog::Confirm(
+                        _("With the scope pointing this close to the pole, calibration accuracy will be degraded and \n"
+                        "Dec compensation will be ineffective. Calibration within 10 degrees of the pole may fail altogether."),

```

Comment:
should we also say "run the Calibration Assistant to improve results" here too? the people who hit this error are the ones who especially need to know!